### ISSUE TITLE
Does SVD_Loss backpropagate when decoder features are detached?

### ISSUE THREAD

**Issue by VGDXHan** [jqtangust/IUF#10]
Hi, thanks for releasing the official implementation of IUF.

I have a question about the implementation of the Semantic Compression Loss / SVD loss.

In `tools/train_val.py`, the decoder features are accumulated as follows:

```python
concatenated_tensor_0 = torch.cat(
    [concatenated_tensor_0, outputs["middle_decoder_feature_0"].clone().detach()],
    dim=0,
)
concatenated_tensor_1 = torch.cat(
    [concatenated_tensor_1, outputs["middle_decoder_feature_1"].clone().detach()],
    dim=0,
)
concatenated_tensor_2 = torch.cat(
    [concatenated_tensor_2, outputs["middle_decoder_feature_2"].clone().detach()],
    dim=0,
)

loss_svd = SVDLoss(
    concatenated_tensor_0,
    concatenated_tensor_1,
    concatenated_tensor_2,
)

if skip == True:
    loss += 10 * loss_svd
```

However, since the three middle decoder features are detached before being passed into SVDLoss, it seems that loss_svd cannot backpropagate gradients to the reconstruction network.
